core/test/surplus-test.ts

```diff
@@ -0,0 +1,105 @@
+
+import { SurplusAuctionActive } from '../src/types';
+import hre from 'hardhat';
+
+const REMOTE_RPC_URL = process.env.REMOTE_RPC_URL;
+const HARDHAT_PRIVATE_KEY = '0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80'; // deterministic private key from hardhat.
+const HARDHAT_FORK_BLOCK_NUMBER = 14078339;
+
+async function createWalletFromPrivateKey(privateKey: string, network: string) {
+    setSigner(network, createSigner(network, privateKey));
+    const signer = await getSigner(network);
+    return await signer.getAddress();
+}
+
+describe('Surplus Auction', () => {
+    before(async () => {
+        const local_rpc_url = process.env.LOCAL_RPC_URL || 'http://localhost:8545';
+        await setupRpcUrlAndGetNetworks(local_rpc_url);
+    });
+    beforeEach(async () => {
+        await hre.network.provider.request({
+            method: 'hardhat_reset',
+            params: [
+                {
+                    forking: {
+                        jsonRpcUrl: REMOTE_RPC_URL,
+                        blockNumber: HARDHAT_FORK_BLOCK_NUMBER,
+                    },
+                },
+            ],
+        });
+    });
+    it('fetches active auctions', async () => {
+        const auctions = await fetchActiveSurplusAuctions('custom');
+        expect(auctions.length).to.equal(5);
+        expect(auctions[0].id).to.equal(2328);
+        expect(auctions[0].state).to.equal('have-bids');
+    });
+    it('participates in active auction', async () => {
+        const address = await createWalletFromPrivateKey(HARDHAT_PRIVATE_KEY, 'custom');
+        await setAllowanceAmountMKR('custom', address, '20');
+        await swapToMKR('custom', 20, 20);
+        await bidToSurplusAuction('custom', 2328, '20');
+        const auctions = await fetchActiveSurplusAuctions('custom');
+        const currentAuction = auctions[0] as SurplusAuctionActive;
+        expect(currentAuction.id).to.equal(2328);
+        expect(currentAuction.bidAmountMKR.eq(20)).to.be.true;
+    });
+    it('collects the concluded auction', async () => {
+        const auctionsBeforeCollection = await fetchActiveSurplusAuctions('custom');
+        expect(auctionsBeforeCollection.length).to.equal(5);
+        expect(auctionsBeforeCollection[1].id).to.equal(2327);
+        expect(auctionsBeforeCollection[1].state).to.equal('ready-for-collection');
+
+        await collectSurplusAuction('custom', auctionsBeforeCollection[1].id);
+
+        const auctionAfterCollection = await fetchSurplusAuctionByIndex('custom', 2327);
+        expect(auctionAfterCollection.id).to.equal(2327);
+        expect(auctionAfterCollection.state).to.equal('collected');
+    });
+    it('restarts the auction', async () => {
+        await hre.network.provider.request({
+            method: 'hardhat_reset',
+            params: [
+                {
+                    forking: {
+                        jsonRpcUrl: REMOTE_RPC_URL,
+                        blockNumber: 14078338,
```

Comment:
Should it be `HARDHAT_FORK_BLOCK_NUMBER`?

Reply:
`HARDHAT_FORK_BLOCK_NUMBER - 1`** :)